lightning/src/ln/channelmanager.rs

```diff
@@ -3478,6 +3504,73 @@ impl<ChanSigner: ChannelKeys, M: Deref + Sync + Send, T: Deref + Sync + Send, K:
 	}
 }
 
+/// Used to signal to the ChannelManager persister that the manager needs to be re-persisted to
+/// disk/backups, through `wait_timeout`.
+pub struct PersistenceNotifier {
+	/// Used to signal to the ChannelManager persister that the manager needs to be re-persisted to
+	/// disk/backups. Users won't access the persistence_lock directly, but rather wait on its bool
+	/// using wait_timeout.
+	#[cfg(any(test, feature = "_test_utils"))]
+	pub persistence_lock: Arc<(Mutex<bool>, Condvar)>,
+	#[cfg(not(any(test, feature = "_test_utils")))]
+	persistence_lock: Arc<(Mutex<bool>, Condvar)>,
+}
+
+impl PersistenceNotifier {
+	fn new() -> Self {
+		Self {
+			persistence_lock: Arc::new((Mutex::new(false), Condvar::new())),
+		}
+	}
+
+	/// Blocks until ChannelManager needs to be persisted or a timeout is reached. If no timeout is
+	/// provided, this function will block indefinitely until persistence is needed. It returns a bool
+	/// indicating whether persistence is necessary (which will always be true if max_wait is None).
+	#[cfg(feature = "allow_wallclock_use")]
```

Comment:
Should there be a version that doesn't support timeouts that doesn't need the feature flag? Not sure it would be really useful, but its something.
In any case, I think we need flags to tell docs.rs to build with this feature, and also documentation noting this feature is required for this fn.

Reply:
Added both!